api/app/src/Controller/UserResearchController.php

```diff
@@ -82,4 +82,24 @@ public function getAll(Request $request)
             throw new \RuntimeException(sprintf('There was a problem getting user research responses: %s', $e->getMessage()), Response::HTTP_BAD_REQUEST);
         }
     }
+
+    /**
+     * @Route("/user-research/{id}", name="user_research_find_by_id", methods={"PUT"})
+     *
+     * @Security("is_granted('ROLE_SUPER_ADMIN')")
+     */
+    public function findByUserIdAndNullify(int $id): void
+    {
+        $this->userResearchResponseRepository->findByUserId($id);
+    }
+
+    /**
+     * @Route("/user-research/delete", name="delete_user_research", methods={"DELETE"})
+     *
+     * @Security("is_granted('ROLE_SUPER_ADMIN')")
+     */
+    public function deleteNullifiedUserId(): void
```

Comment:
We need to some functional tests around this endpoint. Every release we do, should be confirming that only a super admin can ever access this. We do have controller tests on the user endpoints that also have a deletion action,

I'd go as far as suggesting this is moved to a daily job rather than having an endpoint, as once the FK has been removed. Exactly when the entry is deleted isn't a point of concern long as we do delete the entry.